src/ethereum_test_types/types.py

```diff
@@ -1248,12 +1248,18 @@ def __init__(
     ):
         """
         Initializes the requests object.
+
+        If `requests_lists` is given as argument, assumes they are sorted by request type
+        ascending, starting from 0 and contiguous - including empty entries.
         """
         if requests_lists is not None:
             assert len(requests) == 0, "requests must be empty if list is provided"
             self.requests_list = []
-            for requests_list in requests_lists:
-                self.requests_list.append(requests_list_to_bytes(requests_list))
+            for type, requests_list in enumerate(requests_lists):
+                if requests_list:
+                    self.requests_list.append(
+                        Bytes(bytes([type]) + requests_list_to_bytes(requests_list))
```

Comment:
This still breaks our test generation because this is the code path we use to process the list returned by the t8n tool, and the list returned by t8n (correctly) already contains the type byte at the beginning of each request.

I think the interface could definitely be improved, and it could be that we need a different parameter for this use case in this PR, and a different one for the values returned by t8n.

It would also help me to better understand this use case with an example test that causes an issue when wrapped with an EOF container.